pkg/app/api/api/web_api.go

```diff
@@ -316,6 +331,26 @@ func (a *WebAPI) getPiped(ctx context.Context, pipedID string) (*model.Piped, er
 	return piped, nil
 }
 
+// pipedBelongsToProject checks if the given piped belongs to the given project.
+// It gives back error unless the piped belongs to the project.
+func (a *WebAPI) pipedBelongsToProject(ctx context.Context, pipedID, projectID string) error {
```

Comment:
nit: Prefix this by `validate...`.

We can have many kinds of validation: payload validation, auth validation... I think prefixing them makes the code become clearer.